sky/packages/sky/lib/src/http/mojo_client.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library base_client;
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:mojo/core.dart' as mojo;
+import 'package:mojo/mojo/url_request.mojom.dart' as mojo;
+import 'package:mojo/mojo/url_response.mojom.dart' as mojo;
+import 'package:mojo_services/mojo/network_service.mojom.dart' as mojo;
+import 'package:mojo_services/mojo/url_loader.mojom.dart' as mojo;
+
+import 'package:flutter/src/services/shell.dart';
+
+import 'client.dart';
+import 'response.dart';
+
+mojo.NetworkServiceProxy _initNetworkService() {
+  mojo.NetworkServiceProxy networkService = new mojo.NetworkServiceProxy.unbound();
+  shell.connectToService("mojo:authenticated_network_service", networkService);
+  return networkService;
+}
+
+final mojo.NetworkServiceProxy _networkService = _initNetworkService();
+
+/// A `mojo`-based HTTP client
+class MojoClient implements Client {
+  /// Sends an HTTP HEAD request with the given headers to the given URL, which
+  /// can be a [Uri] or a [String].
+  ///
+  /// For more fine-grained control over the request, use [send] instead.
```

Comment:
Do we need all these doc comments?  Presumably the docs will just be in the Client class.